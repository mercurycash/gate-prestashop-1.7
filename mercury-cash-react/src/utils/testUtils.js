import { render } from "@testing-library/react";
import { Provider } from "outstated";
import { store } from "../state/state";

export const customRender = (ui) => {
  return render(
    <Provider stores={ [ store ] }>{ ui }</Provider>
  )
}
