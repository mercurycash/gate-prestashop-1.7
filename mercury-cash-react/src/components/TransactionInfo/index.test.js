import { TransactionInfo } from "./index";
import { customRender } from "../../utils/testUtils";

const children = (
  <>
    <li>
      <strong>Amount to be Paid USD:</strong> <span>$50.00 USD</span>
    </li>
    <li>
      <strong>Exchange rate:</strong> <span>1,182.77 USD</span>
    </li>
    <li>
      <strong>Total:</strong> <span>0.04230942 ETH</span>
    </li>
  </>
)

describe('<WalletAddress/>', () => {
  const spy = jest.spyOn(global.console, "error")

  it('Render Loader Component', async () => {
    const { container } = customRender(<TransactionInfo>{ children }</TransactionInfo>)
    expect(container).toMatchSnapshot()
    expect(spy).not.toHaveBeenCalled()
  })
})