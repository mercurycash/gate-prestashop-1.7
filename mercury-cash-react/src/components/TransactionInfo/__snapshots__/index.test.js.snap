// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WalletAddress/> Render Loader Component 1`] = `
<div>
  <ul
    class="mercury-cash__pain-info"
  >
    <li>
      <strong>
        Amount to be Paid USD:
      </strong>
       
      <span>
        $50.00 USD
      </span>
    </li>
    <li>
      <strong>
        Exchange rate:
      </strong>
       
      <span>
        1,182.77 USD
      </span>
    </li>
    <li>
      <strong>
        Total:
      </strong>
       
      <span>
        0.04230942 ETH
      </span>
    </li>
  </ul>
</div>
`;
